src/pages/leaderboard/subcomponents/LeaderboardExportButton.tsx

```diff
@@ -15,60 +18,50 @@ type Props = {
 };
 
 const LeaderboardExportButton: React.FC<Props> = ({ type, contest, contestID }) => {
-  // Retrieve relevant leaderboard data
-  const [exportRequested, setExportRequest] = useState(false);
-  const dispatch = useDispatch();
-  const selectData = (type: string) => {
-    switch (type) {
-      case 'overall':
-        return (store: { leaderboard: { userXp: any } }) => store.leaderboard.userXp;
-      case 'score':
-        return (store: { leaderboard: { contestScore: any } }) => store.leaderboard.contestScore;
-      default:
-        return (store: { leaderboard: { contestPopularVote: any } }) =>
-          store.leaderboard.contestPopularVote;
-    }
-  };
-
-  const selector = useMemo(() => selectData(type), [type]);
-  const data = useTypedSelector(selector);
-
-  const visibleEntries = Number.MAX_SAFE_INTEGER;
+  const role = useTypedSelector(store => store.session.role);
 
-  const onExportClick = () => {
-    // Dispatch relevant request
-    if (type == 'overall') dispatch(LeaderboardActions.getAllUsersXp());
-    else if (type == 'score')
-      dispatch(LeaderboardActions.getAllContestScores(contestID as number, visibleEntries));
-    else
-      dispatch(LeaderboardActions.getAllContestPopularVotes(contestID as number, visibleEntries));
-    setExportRequest(true);
-  };
+  const onExportClick = async () => {
+    const accessToken = store.getState().session.accessToken!;
+    const refreshToken = store.getState().session.refreshToken!;
```

Comment:
Why not destructure from Redux `useTypedSelector` earlier?

Reply:
It's a one-time operation which would unlikely need keeping track of the tokens as state, but if you think its better we can change that.